packages/@aws-cdk/aws-kinesis/lib/stream.ts

```diff
@@ -325,21 +325,13 @@ export class Stream extends StreamBase {
     const encryptionType = props.encryption ??
       (props.encryptionKey ? StreamEncryption.KMS : StreamEncryption.UNENCRYPTED);
 
-    // if encryption key is set, encryption must be set to KMS.
```

Comment:
What happens if `encryption` is explicitly set to `UNENCRYPTED` and `encryptionKey` is set. I believe we should throw an error in that case.

Reply:
good point. I think the error message needs fixing rather than the removal of this condition (and a test).

Reply:
good point, I'll add a test for now. maybe it makes sense to improve the wording on the error but I'm okay with it